app/controllers/authenticators_controller.rb

```diff
@@ -0,0 +1,102 @@
+# frozen_string_literal: true
+
+# REST endpoints for persisting authenticator configuration details in Conjur
+class AuthenticatorsController < RestController
+  include FindResource
+  include AuthorizeResource
+  
+  before_action :current_user
+  before_action :find_or_create_conjur_policy_branch
+  before_action :parse_request_body
+
+  rescue_from Sequel::UniqueConstraintViolation, with: :concurrent_load
+
+  def persist_auth
+    policy_exists = policy_resource_exists(auth_policy_branch)
+    policy_details = Authentication::PersistAuthFactory.new_from_authenticator(authenticator).(
+      conjur_account: params[:account],
+      service_id: service_id,
+      resource: @conjur_policy_branch,
+      current_user: current_user,
+      client_ip: request.ip,
+      request_data: @request_data
+    )
+    render(body: policy_text(policy_details), content_type: "text/yaml", status: policy_exists ? :ok : :created)
+  end
+
+  def persist_gcp_auth
+    raise UnprocessableEntity, "GCP authenticatior takes no arguments" unless @request_data.empty?
+
+    policy_exists = policy_resource_exists(auth_policy_branch)
+    policy_details = Authentication::PersistAuthFactory.new_from_authenticator(authenticator).(
+      conjur_account: params[:account],
+      service_id: service_id,
+      resource: @conjur_policy_branch,
+      current_user: current_user,
+      client_ip: request.ip,
+      request_data: {}
+    )
+    render(body: policy_text(policy_details), content_type: "text/yaml", status: policy_exists ? :ok : :created)
+  end
+
+  protected
+
+  def find_or_create_conjur_policy_branch
+    @conjur_policy_branch = ::Resource[conjur_policy_branch_name] || create_conjur_policy_branch
+  end
+
+  private
+
+  def create_conjur_policy_branch
+    ::Role.create(role_id: conjur_policy_branch_name)
+    ::Resource.create(resource_id: conjur_policy_branch_name, owner: ::Role["#{account}:user:admin"])
+  end
+
+  def conjur_policy_branch_name
+    "#{account}:policy:conjur"
+  end
+
+  def policy_text(loaded_policy_details)
+    loaded_policy_details[:policy].values[:policy_text]
+  end
+
+  def policy_resource_exists(policy_branch_name)
```

Comment:
in ruby, functions which result in a boolean value typically get a question mark:
```ruby
policy_resource_exists?
```
to improve readability.